torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -87,26 +87,27 @@ Reducer::Reducer(
       for (size_t variable_index = 0; variable_index < variable_count;
            variable_index++) {
         auto& variable = replicas_[replica_index][variable_index];
-        const auto index = VariableIndex{
-            .replica_index = replica_index,
-            .variable_index = variable_index,
-        };
+        const auto index = VariableIndex(replica_index, variable_index);
 
         // The gradient accumulator function is lazily initialized once.
         // Therefore we can use its presence in the autograd graph as
         // evidence that the parameter has participated in an iteration.
         auto grad_accumulator =
             torch::autograd::impl::grad_accumulator(variable);
 
+#ifndef _WIN32
         using torch::distributed::autograd::ThreadLocalDistAutogradContext;
+#endif
         // Hook to execute after the gradient accumulator has executed.
         hooks_.emplace_back(
             grad_accumulator->add_post_hook(
                 torch::make_unique<torch::autograd::utils::LambdaPostHook>(
                     [=](const torch::autograd::variable_list& outputs,
                         const torch::autograd::variable_list& /* unused */) {
+#ifndef _WIN32
```

Comment:
@pritamdamania87 please let us know if there are concerns about this change.

Reply:
I guess we're doing this since we're only supporting DDP on windows and not the RPC framework yet? If so, we should update our docs to clearly mention this.

If we try to import the RPC package on windows do we get a reasonable error message? We shouldn't be in a scenario where a user tries to use DDP + RPC on Windows and it silently doesn't work.

Reply:
> I guess we're doing this since we're only supporting DDP on windows and not the RPC framework yet? If so, we should update our docs to clearly mention this.

Yes

> If we try to import the RPC package on windows do we get a reasonable error message? We shouldn't be in a scenario where a user tries to use DDP + RPC on Windows and it silently doesn't work.

Good point! cc @gunandrose4u Let's make sure importing RPC on windows throws clear error message.

Reply:
Or is RPC already unavailable on Windows due to this? 

https://github.com/pytorch/pytorch/blob/721cfbf8425cf2c1dc5e27d1332e32e1a42ef541/torch/distributed/rpc/__init__.py#L14-L15